File: src/Z.Net/System.Net.WebResponse/WebResponse.ReadToEnd.cs
// Description: C# Extension Methods Library to enhances the .NET Framework by adding hundreds of new methods. It drastically increases developers productivity and code readability. Support C# and VB.NET
// Website & Documentation: https://github.com/zzzprojects/Z.ExtensionMethods
// Forum: https://github.com/zzzprojects/Z.ExtensionMethods/issues
// License: https://github.com/zzzprojects/Z.ExtensionMethods/blob/master/LICENSE
// More projects: http://www.zzzprojects.com/
// Copyright © ZZZ Projects Inc. 2014 - 2016. All rights reserved.
using System.IO;
using System.Net;

public static partial class Extensions
{
    /// <summary>
    ///     A WebResponse extension method that reads the response stream to the end.
    /// </summary>
    /// <param name="this">The @this to act on.</param>
    /// <returns>The response stream as a string, from the current position to the end.</returns>
    public static string ReadToEnd(this WebResponse @this)
    {
        using (Stream stream = @this.GetResponseStream())
        {
            using (var reader = new StreamReader(stream))
            {
                return reader.ReadToEnd();
            }
        }
    }
}

File: src (VB.NET)/Z.ExtensionMethods.VB/Z.Net/System.Net.WebResponse/WebResponse.ReadToEnd.vb

' Copyright (c) 2015 ZZZ Projects. All rights reserved
' Licensed under MIT License (MIT) (https://github.com/zzzprojects/Z.ExtensionMethods)
' Website: http://www.zzzprojects.com/
' Feedback / Feature Requests / Issues : http://zzzprojects.uservoice.com/forums/283927
' All ZZZ Projects products: Entity Framework Extensions / Bulk Operations / Extension Methods /Icon Library

Imports System.IO
Imports System.Net

Public Module Extensions_803

	''' <summary>
	'''     A WebResponse extension method that reads the response stream to the end.
	''' </summary>
	''' <param name="this">The @this to act on.</param>
	''' <returns>The response stream as a string, from the current position to the end.</returns>
	<System.Runtime.CompilerServices.Extension> _
	Public Function ReadToEnd(this As WebResponse) As String
		Using stream As Stream = this.GetResponseStream()
			Using reader = New StreamReader(stream)
				Return reader.ReadToEnd()
			End Using
		End Using
	End Function
End Module


